### PR TITLE
placement: give default 2 followers for non-sugar syntax

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: subtask of #18030 

Problem Summary: It seems still counterintuitive that non-sugar syntax will not give default of `1 leader + 2 followers`, while sugar syntax does give 2 default followers. Make it consistent.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]
LGTM

File: ddl/placement/bundle.go [hunk 1]
```diff
@@ -217,6 +205,10 @@ func newBundleFromOptions(options *model.PlacementSettings) (bundle *Bundle, err
 		isSyntaxSugar = false
 	}
 
+	if options.Followers == 0 {
```
Comment (lcwangchao) [inline ddl/placement/bundle.go:208]: I found a new problem... `create table t1(id int) followers=0` still create 2 followers...

Comment (lcwangchao) [reply ddl/placement/bundle.go:208]: But anyway, we can handle it later to add a variable `IsFollowerSet` in PlacementSettings ...

**Review by lcwangchao — APPROVED** [review]

File: ddl/placement/bundle.go (hunk 1 above)
Comment (xhebox) [reply ddl/placement/bundle.go:208]: It is forbidden for now. Let us leave the problem for future.

**Pull request merged**